### PR TITLE
Fixes warnings.

### PR DESCRIPTION
**Pull request by damncabbage**
I'm in the process of adding a Flow language, but stuff like pscid is spitting out warnings on compile. I've gone through and fixed them, to avoid the _get warnings_ -> _read README + build script_ -> _reconfigure tools to turn off warnings_  progression when other people looking to contribute; I hope they match your style well enough.

### CODE REVIEW

File: src/IRGraph.purs [hunk 1]
```diff
@@ -377,29 +383,28 @@ forUnion_ (IRUnionRep { primitives, arrayType, classRef, mapType }) f = do
...
+    pure L.Nil
+        >>= mapGeneral mapType IRMap
+        >>= mapGeneral classRef IRClass
+        >>= mapGeneral arrayType IRArray
+        >>= mapPrimitive irUnion_String IRString
+        >>= mapPrimitive irUnion_Bool IRBool
+        >>= mapPrimitive irUnion_Double IRDouble
+        >>= mapPrimitive irUnion_Integer IRInteger
+        >>= mapPrimitive irUnion_Null IRNull
+        >>= mapPrimitive irUnion_Nothing IRNothing
```
Comment (damncabbage) [inline src/IRGraph.purs]: This slab of changes was because I was getting a bunch of shadow warnings for `l`.

**Comment by schani** [comment]
That's awesome, thank you! I've wanted to fix warning for a long time, I was just too lazy to do it.

Would you like to join us on Slack for discussion?  I can email you an invite if you'd like.

**Comment by damncabbage** [comment]
Sure! I'd appreciate that. (I'm on the wealljs and functionalprogramming Slacks already, in case either of those are the Slack you mean.)

**Comment by schani** [comment]
Invite emailed.

**Pull request merged**